### PR TITLE
Migrate test classes to AssertJ

### LINKED ISSUE #1721: @RetryableListener - Convert Tests to Use AssertJ
For consistency.

(I didn't notice during code review).

cc/ @tomazfernandes

### PR DESCRIPTION
**Pull request by aravindram87**
Migrated all test classes from JUnit5 asserts to use AssertJ library.

Fixes #1721

### CODE REVIEW

File: spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicBootstrapperTests.java [hunk 1]
```diff
@@ -66,12 +66,14 @@
 
 	@Test
 	void shouldThrowIfACDoesntImplementInterfaces() {
-		assertThrows(IllegalStateException.class, () -> new RetryTopicBootstrapper(wrongApplicationContext, beanFactory));
+		assertThatExceptionOfType(IllegalStateException.class)
+				.isThrownBy(() -> new RetryTopicBootstrapper(wrongApplicationContext, beanFactory));
 	}
```
Comment (garyrussell) [inline spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicBootstrapperTests.java:71]: We prefer to use `assertThatIllegalStateException()` and `assertThatIllegalArgumentException()` for these.

**Review by garyrussell — COMMENTED** [review]
Thanks for the PR; just one comment.

Are you planning to fix the remaining test classes?

No worries if not; we can do it.

Thanks again.

**Comment by aravindram87** [comment]
Sure. I can fix the remaining test classes. Just to confirm, all test classes need to be moved over to AssertJ?

**Comment by garyrussell** [comment]
Yes, thanks; I fixed the checkstyle.xml to detect JUnit5 asserts (it previously only detected JUnit4).

If you rebase your branch against master, you'll see the remaining errors:

```
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaConsumerBackoffManagerTest.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaConsumerBackoffManagerTest.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaConsumerBackoffManagerTest.java:21: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/listener/adapter/KafkaBackoffAwareMessageListenerAdapterTest.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/BackOffValuesGeneratorTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactoryTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactoryTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactoryTests.java:21: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DefaultDestinationTopicProcessorTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DefaultDestinationTopicProcessorTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DestinationTopicContainerTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DestinationTopicContainerTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DestinationTopicPropertiesFactoryTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DestinationTopicPropertiesFactoryTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/DestinationTopicPropertiesFactoryTests.java:21: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/ListenerContainerFactoryConfigurerTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/ListenerContainerFactoryConfigurerTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/ListenerContainerFactoryResolverTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/ListenerContainerFactoryResolverTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/retrytopic/ListenerContainerFactoryResolverTests.java:21: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/support/KafkaStreamBrancherTests.java:19: Please use AssertJ imports. [RegexpSinglelineJava]
[ant:checkstyle] [ERROR] /Users/grussell/Development/spring-kafka/spring-kafka/src/test/java/org/springframework/kafka/support/KafkaStreamBrancherTests.java:20: Please use AssertJ imports. [RegexpSinglelineJava]
```

File: spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicBootstrapperTests.java (hunk 1 above)
Comment (aravindram87) [reply spring-kafka/src/test/java/org/springframework/kafka/retrytopic/RetryTopicBootstrapperTests.java:71]: @garyrussell Fixed.

**Comment by garyrussell** [comment]
Many thanks for the contribution!

**Pull request merged**